middlewares/doctorAuthentication.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken'
+import CreateDoctorModel from "../models/doctormodel.js";
+
+const authenticateDoctor = async (req, res, next) => {
+ 
+
+  try {
+   
+    const decoded = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET);
+    const userId = decoded.id;
+
+    const check= await CreateDoctorModel.findById({_id : userId})
+      if (!check) {
+        return res.status(401).json({ error: 'Authentication failed' });
+      }
+
+      req.user = check;
```

Comment:
here why do you need to set the req.user to check?

Reply:
I learnt that it is necessary to attach it on the request for future use, but I'm going to change it as well